pkg/ingester-rf1/metastore/metastore_state.go

```diff
@@ -9,30 +9,34 @@ import (
 	"github.com/gogo/protobuf/proto"
 	"go.etcd.io/bbolt"
 
+	ingesterrf1 "github.com/grafana/loki/v3/pkg/ingester-rf1"
 	metastorepb "github.com/grafana/loki/v3/pkg/ingester-rf1/metastore/metastorepb"
 )
 
 type metastoreState struct {
 	logger log.Logger
 
-	segmentsMutex sync.Mutex
-	segments      map[string]*metastorepb.BlockMeta
+	active map[string]*metastorepb.BlockMeta
+	marked map[string]*metastorepb.BlockMeta
+	mtx    sync.Mutex
 
-	db *boltdb
+	db    *boltdb
+	store ingesterrf1.Storage
 }
 
 func newMetastoreState(logger log.Logger, db *boltdb) *metastoreState {
 	return &metastoreState{
-		logger:   logger,
-		segments: make(map[string]*metastorepb.BlockMeta),
-		db:       db,
+		logger: logger,
+		active: make(map[string]*metastorepb.BlockMeta),
+		marked: make(map[string]*metastorepb.BlockMeta),
+		db:     db,
 	}
 }
 
 func (m *metastoreState) reset() {
-	m.segmentsMutex.Lock()
-	clear(m.segments)
-	m.segmentsMutex.Unlock()
+	m.mtx.Lock()
+	clear(m.active)
```

Comment:
should this also clear marked or do we not care about those here?

Reply:
Yes, thanks! 😄